utils/subfiling_vfd/h5fuse.sh.in

```diff
@@ -23,15 +24,40 @@ function usage() {
    echo ""
    # Display usage
    echo "Purpose: Combine subfiles into a single HDF5 file. Requires the subfiling
-         configuration file either as a command-line argument, or the script will
+         configuration file either as a command-line argument or the script will
          search for the *.config file in the current directory."
    echo ""
-   echo "usage: h5fuse.sh [-h] [-f filename]"
-   echo "-h            Print this help."
-   echo "-f filename   Subfile configuration file."
+   echo "usage: h5fuse.sh [-f filename] [-h] [-p] [-q] [-r] [-v] "
+   echo "-f filename  Subfile configuration file."
+   echo "-h           Print this help."
+   echo "-q           Quiet all output. [no]"
+   echo "-p           h5fuse.sh is being run in parallel, with more than one rank. [no]"
+   echo "-r           Remove subfiles after being processed. [no]"
+   echo "-v           Verbose output. [no]"
    echo ""
 }
 
+function gen_mpi() {
+
+# Program to determine MPI rank and size if being run in parallel (-p).
+
+cat > "${c_src}" << EOL
+#include <mpi.h>
+#include <stdio.h>
+int main() {
+    MPI_Init(NULL, NULL);
+    int world_size;
+    MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+    int world_rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
+    printf("%d %d", world_rank, world_size);
```

Comment:
Is this program being run on all ranks? If so I'd suggest printing this out on a single rank in case the output would get interleaved and mess things up

Reply:
It is being run on each rank to get the rank id. So each rank would print to standard out. Are you saying that other ranks can see others standard out? How can I run it on one rank to get a rank's id?

Reply:
I just realized the program is essentially being run as a serial program on every rank. Without running through mpiexec with a specified number of procs, wouldn't this program just always return 1 for the comm size and 0 for the rank number?